php/functions.php

```diff
@@ -0,0 +1,250 @@
+ *
+ * @param bool $newAboutMeText outcome of adding text to the database
+ *
+ * @return string display message to say whether the add text has been added to the database
+ */
+function aboutMeSuccess(bool $newAboutMeText) : string {
+    if ($newAboutMeText === false) {
+        return "<p>Please add content of up to 400 characters</p>";
+    } else {
+        return "<p>Your text has been added</p>";
+    }
+}
+
+/**
+ * Returns an array from the database of all lines of text
+ *
+ * @param PDO $db database connection
+ *
+ * @return array of id and content for all lines of the database
+ */
+function getAboutMeTextAndQuote (PDO $db) : array {
+    $query = $db->prepare("SELECT `id`,`content` FROM `about_me` WHERE `deleted` = 0");
+    $query->execute();
+    return $query->fetchAll();
+}
+
+/**
+ * Takes content from the database, shortens the length and inserts into a string of html
+ *
+ * @param array $aboutMeTextAndQuotes from database fetch
+ *
+ * @return string of options for the edit text dropdown
+ */
+function aboutMeTextDropdown (array $aboutMeTextAndQuotes) : string {
+    $dropdown = "";
+    foreach ($aboutMeTextAndQuotes as $aboutMeTextAndQuote) {
+        $shortAboutMe = substr($aboutMeTextAndQuote['content'], 0, 50);
+        $dropdown .= '<option value='. $aboutMeTextAndQuote['id'] . '>' . $shortAboutMe . '</option>';
+    }
+    return $dropdown;
+}
+
+/**
+ * Fetches content from the database to display in the edit box
+ *
+ * @param PDO $db connection to database
+ * @param string $aboutMeDropDownValue variable assigned from post data in form dropdown
+ *
+ * @return array of one content value from the database
+ */
+function getAboutTextToEdit(PDO $db, string $aboutMeDropDownValue) : array {
+    $query = $db->prepare("SELECT `content` FROM `about_me` WHERE `id` = '$aboutMeDropDownValue';");
+    $query->execute();
+    return $query->fetch();
+}
+
+/**
+ * displays the content value fetched from the database inside the edit box
+ *
+ * @param array $aboutTextToEdits content value fetched from the database
+ *
+ * @return string of content from the array; otherwise returns an empty string
+ */
+
+function displayAboutTextToEdit(array $aboutTextToEdit) : string {
+    $result="";
+        if(array_key_exists('content', $aboutTextToEdit)) {
+            $result .= $aboutTextToEdit['content'];
+        } else {
+            $result .="";
+        }
+    return $result;
+}
+
+/**
+ * Update data base with edited content
+ *
+ * @param PDO $db database connection
+ * @param bool returned from the function which checks whether there are characters in the string
+ * @param string $submitEditText edited content from the form
+ * @param string $editId database id saved in the session
+ *
+ * @return bool updates the database with edited content based on id
+ */
+function updateAboutMeQuoteAndText(PDO $db, bool $checkEditText, string $submitEditText, string $editId) :bool {
```

Comment:
function should just do one thing so do the check on checkEditText outside the function elsewhere.

Reply:
Done